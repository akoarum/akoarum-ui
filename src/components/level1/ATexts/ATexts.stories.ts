import { storiesOf } from '@storybook/vue'
import { withKnobs, text } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import ATexts from './ATexts.vue'

storiesOf('level1/ATexts', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { ATexts },
    props: {
      texts: { default: text('テキスト', '吾輩は猫である。名前はまだない。') },
      size: { default: text('フォントサイズ', '16') },
      lineHeight: { default: text('行間', '1.9') }
    },
    template: `
      <ATexts :size="size" :line-height="lineHeight">{{ texts }}</ATexts>
    `
  }), { info: {} })
  .add('Caution', () => ({
    components: { ATexts },
    props: {
      texts: { default: text('テキスト', '吾輩は猫である。名前はまだない。') },
      size: { default: text('フォントサイズ', '16') },
      lineHeight: { default: text('行間', '1.9') }
    },
    template: `
      <ATexts :size="size" :line-height="lineHeight" caution>{{ texts }}</ATexts>
    `
  }), { info: {} })
  .add('Note', () => ({
    components: { ATexts },
    props: {
      texts: { default: text('テキスト', '吾輩は猫である。名前はまだない。') },
      size: { default: text('フォントサイズ', '16') },
      lineHeight: { default: text('行間', '1.9') }
    },
    template: `
      <ATexts :size="size" :line-height="lineHeight" note>{{ texts }}</ATexts>
    `
  }), { info: {} })
