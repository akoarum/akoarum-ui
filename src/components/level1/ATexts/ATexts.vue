<template>
  <p :style="styled" :class="styledClasses" class="aTexts"><slot /></p>
</template>

<script lang="ts">
import { defineComponent, computed } from '@vue/composition-api'

interface Props {
  size: string
  lineHeight: string
  caution?: boolean
  note?: boolean
}

export default defineComponent({
  props: {
    size: { type: String, default: '16' },
    lineHeight: { type: String, default: '1.9' },
    caution: { type: Boolean },
    note: { type: Boolean }
  },
  setup(props: Props) {
    const styled = computed<{ [key: string]: string }>(() => {
      return {
        '--size': `${props.size}px`,
        '--lineHeight': props.lineHeight
      }
    })
    const styledClasses = computed<string[]>(() => {
      const classes: string[] = []
      if (props.caution) classes.push('-caution')
      if (props.note) classes.push('-note')
      return classes
    })

    return { styled, styledClasses }
  }
})
</script>

<style lang="scss" scoped>
.aTexts {
  --size: 16px;
  --lineHeight: 1.9;
  font-size: var(--size);
  line-height: var(--lineHeight);
  white-space: pre-line;

  &.-caution {
    color: $color-caution;
  }
  &.-note {
    color: #888;
  }
}
</style>
