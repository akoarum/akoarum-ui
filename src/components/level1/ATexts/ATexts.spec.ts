import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import ATexts from './ATexts.vue'

describe('ATexts', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(ATexts, {
      propsData: {},
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('見た目の制御', () => {
    it('styled: propsで指定したサイズ・行間で表示できる', () => {
      wrapper.setProps({ size: '18', lineHeight: '2' })
      expect(wrapper.vm.styled).toEqual({
        '--size': '18px',
        '--lineHeight': '2'
      })
    })

    it('styledClasses: propsで指定した色に変更できる', () => {
      wrapper.setProps({ caution: true })
      expect(wrapper.vm.styledClasses).toEqual(['-caution'])
      wrapper.setProps({ caution: false, note: true })
      expect(wrapper.vm.styledClasses).toEqual(['-note'])
    })
  })
})
