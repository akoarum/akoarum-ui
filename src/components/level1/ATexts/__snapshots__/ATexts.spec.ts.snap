// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATexts initialized snapshot 1`] = `
<p
  class="aTexts"
  style="--size: 16px; --lineHeight: 1.9;"
/>
`;
